src/Aspire.Dashboard/Utils/FormatHelpers.cs

```diff
@@ -26,30 +32,43 @@ static string GetLongTimePatternWithMilliseconds()
     [GeneratedRegex("(:ss|:s)")]
     private static partial Regex MatchSecondsInTimeFormatPattern();
 
-    private static readonly string s_longTimePatternWithMilliseconds = GetLongTimePatternWithMilliseconds();
-    private static readonly string s_shortDateLongTimePatternWithMilliseconds = DateTimeFormatInfo.CurrentInfo.ShortDatePattern + " " + s_longTimePatternWithMilliseconds;
+    private static MillisecondFormatStrings GetMillisecondFormatStrings(CultureInfo cultureInfo)
+    {
+        return s_formatStrings.GetOrAdd(cultureInfo.Name, static name =>
+        {
+            var c = CultureInfo.GetCultureInfo(name);
+            var longTimePatternWithMilliseconds = GetLongTimePatternWithMillisecondsCore(c);
+            return new MillisecondFormatStrings(longTimePatternWithMilliseconds, c.DateTimeFormat.ShortDatePattern + " " + longTimePatternWithMilliseconds);
+        });
```

Comment:
We can avoid looking up the culture by passing on the stack by threading it through via `TArg`. The callback delegate is still static/non-allocated.

```suggestion
        return s_formatStrings.GetOrAdd(cultureInfo.Name, static (name, cultureInfo) =>
        {
            var longTimePatternWithMilliseconds = GetLongTimePatternWithMillisecondsCore(cultureInfo);
            return new MillisecondFormatStrings(longTimePatternWithMilliseconds, cultureInfo.DateTimeFormat.ShortDatePattern + " " + longTimePatternWithMilliseconds);
        },
        cultureInfo);
```

Reply:
The reason I chose the name as the key is it is possible to create custom cultures. I didn't want the collection to grow unbounded. But on the other hand, a custom culture with new format values that uses the same name will be lost. I doubt that situation will ever exist in the dashboard.

`CultureInfo.GetCultureInfo(name);` inside the func ensures that the registered values for that name are always used. And it should be called just a couple of times for the lifetime of the app so perf doesn't matter.

There isn't a perfect solution here. I'll add this as a comment to explain the tradeoff.